### PR TITLE
refactor(db): Convert PasswordResetDAO to be able to use query_as!

### PR DESCRIPTION
**Pull request by boxbeam**

### CODE REVIEW

**Review by wsxiaoys — COMMENTED** [review]
Otherwise LGTM

File: ee/tabby-webserver/src/service/auth.rs [hunk 1]
```diff
@@ -247,7 +247,9 @@ impl AuthenticationService for AuthenticationServiceImpl {
             password_hash(password).map_err(|_| PasswordResetError::Unknown)?;
 
         let user_id = self.db.verify_password_reset(code).await?;
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/auth.rs:249]: verify_password_reset should returns i64 as well?

Comment (boxbeam) [reply ee/tabby-webserver/src/service/auth.rs:249]: Already does? https://github.com/TabbyML/tabby/pull/1469/commits/d1aae1c8c444fac07698939ebf7b27ff2746c3f8

Comment (wsxiaoys) [reply ee/tabby-webserver/src/service/auth.rs:249]: then why it need another cast in 251?

Comment (boxbeam) [reply ee/tabby-webserver/src/service/auth.rs:249]: That's redundant, I suppose

**Pull request merged**